test/unit/gateways/cashnet_test.rb

```diff
@@ -22,6 +22,21 @@ def test_successful_purchase
     assert_equal '1234', response.authorization
   end
 
+  def test_successful_purchase_with_multiple_items
+    @gateway.expects(:ssl_post).returns(successful_purchase_response)
+    options = { item_codes: { item_code: 'APPFEE', item_code2: 'LOBSTER', item_code3: 'CODES', amount2: 1234, amount3: 4321 } }
+
+    response = stub_comms(@gateway, :ssl_request) do
+      @gateway.purchase(@amount, @credit_card, options)
+    end.check_request do |_method, _endpoint, data, _headers|
+      assert_match('itemcode=FEE&itemcode2=LOBSTER&itemcode3=CODES&amount=1.00&amount2=12.34&amount3=43.21', data)
```

Comment:
NICE

Reply:
@dsmcclain ya know, it's funny you commented here, but just now I was messing around with this test, and noticed that this `assert_match` was passing even when it shouldn't. Here's what I have:

```
end.check_request do |_method, _endpoint, data, _headers|
      assert_match('THIS SHOULD NOT MATCH', data)
    end.respond_with(successful_purchase_response)
```

and it's STILL WORKING?!

Can't figure out why this is the case... any ideas?

Reply:
Nice catch @peteroas!

I think the reason is line 26: `@gateway.expects(:ssl_post).returns(successful_purchase_response)`.

I think because of the way stub_comms is written, stubbing `ssl_post` like this means that `respond_with` on line 33 never gets called, which means that the block passed in `check_request` never actually gets evaluated, which means that we never hit your assertion on line 32. (Pretty sure that's the sequence of events. `comm_stub.rb` is not easy to read.)

In any case, you should be able to actually test the request if you remove line 26.